### PR TITLE
Az4669 feat monorepo tagging

### PR DESCRIPTION
**Pull request by davelanglois-ssc**
PR to merge into the monorepo branch. This branch will be used until we are ready to release the whole monorepo solution.

This PR implements the tagging solution when using monorepos. It completely overwrites the original tagging pipeline/script.

It also include the initial modification to the `hydrate.sh` script that are required to have the monorepo POC functional. @alaincormier-ssc is working on the improved version of that hydrate script. Some linting errors are present in that file. I suggest you don't put too much effort reviewing it. :)

a demo is scheduled on april 18th

### CODE REVIEW

File: scripts/cicd/version-tagging.sh [hunk 1]
```diff
@@ -1,106 +1,161 @@
...
+              version=$(echo "$version" | awk -F. '{$(NF-2)++;$(NF-1)=0;$NF=0;print $0}' OFS=.)
+              ;;
+          *fix!:*)
+              print_success "prefix 'fix!:' found"
+              # The awk command uses the same format as feat!
+              version=$(echo "$version" | awk -F. '{$(NF-2)++;$(NF-1)=0;$NF=0;print $0}' OFS=.)
+              ;;
+          *doc:*)
+              print_success "prefix 'doc:' found"
+              print_info "removing previous tag"
```
Comment (alaincormier-ssc) [inline scripts/cicd/version-tagging.sh:139]: `doc` should probably do nothing?
treating tags as immutable

Comment (davelanglois-ssc) [reply scripts/cicd/version-tagging.sh:139]: I guess what I wanted to do with this is that when you look at the files for a particular tag, you are also getting the latest documentation....knowing that changing the tag will force a reconcile....we can discuss it at the demo

**Review by lucstjean-ssc — APPROVED** [review]
/lgtm

File: scripts/cicd/version-tagging.sh [hunk 2]
```diff
@@ -1,106 +1,161 @@
...
+print_info "doc: which represents an update to documentation won't modify the version but will move the tag to the new commit."
+print_info "commit message not following this convention correlates to a SemVer patch."
+print_info "-----------------------------------"
+
+print_info "git status"
+git status
+print_info "-----------------------------------"
+
+print_info "source version"
+echo "$BUILD_SOURCEVERSION"
```
Comment (alaincormier-ssc) [inline scripts/cicd/version-tagging.sh:38]: maybe we can add a TODO reminder to handle GitHub Actions' equivalent if we want to keep compatibility?

File: scripts/cicd/version-tagging.sh (hunk 1 above)
Comment (alaincormier-ssc) [reply scripts/cicd/version-tagging.sh:139]: OK, will do

Comment (davelanglois-ssc) [reply scripts/cicd/version-tagging.sh]: doc is now configured to do nothin

File: scripts/cicd/version-tagging.sh (hunk 2 above)
Comment (davelanglois-ssc) [reply scripts/cicd/version-tagging.sh]: this section is no longer required because the script now performs a git fetch --unshallow

**Comment by davelanglois-ssc** [comment]
comments have been addressed. @alaincormier-ssc  please resolve conversation if satisfied

**Review by alaincormier-ssc — APPROVED** [review]

**Pull request merged**